dbsnap_verify/rds_funcs.py

```diff
@@ -0,0 +1,267 @@
+    Returns database description document or None.
+    Args:
+        session (:class:`boto.rds2.layer1.RDSConnection`): The RDS api
+            connection where the database is located.
+        db_id (str): The RDS database instance identifier.
+    Returns:
+        dictionary: description of RDS database instance
+    """
+    try:
+        return session.describe_db_instances(
+            DBInstanceIdentifier=db_id
+        )['DBInstances'][0]
+    except session.exceptions.DBInstanceNotFoundFault:
+        return None
+
+
+def get_database_events(session, db_id, event_catagories=None, duration=1440):
+    if not event_catagories:
+        event_catagories = []
+    return session.describe_events(
+        SourceIdentifier=db_id,
+        SourceType='db-instance',
+        EventCategories = event_catagories,
+        Duration = duration,
+    )['Events']
+
+
+def rds_event_messages(session, db_id, event_catagories=None, duration=1440):
+    events = get_database_events(session, db_id, event_catagories, duration)
+    return [i['Message'] for i in events]
+
+
+def modify_db_instance_for_verify(session, db_id, sg_ids):
+    """Modify RDS DB Instance to allow connections.
+    Args:
+        session (:class:`boto.rds2.layer1.RDSConnection`): The RDS api
+            connection where the database is located.
+        db_id (str): The RDS database instance identifier to reset.
+    Returns:
+        str: new raw password
+    """
+    # 16 chars was an arbitrary choice.
+    new_password = generate_password(16)
+    session.modify_db_instance(
+        ApplyImmediately=True,
+        DBInstanceIdentifier=db_id,
+        VpcSecurityGroupIds=sg_ids,
+        BackupRetentionPeriod=0,
+        MasterUserPassword=new_password,
+    )
+    return new_password
+
+
+def make_tag_dict(tag_list):
+    """Returns a dictionary of existing tags.
+    Args:
+        tag_list (list): a list of tag dicts.
+    Returns:
+        dict: A dictionary where tag names are keys and tag values are values.
+    """
+    return {i["Key"]: i["Value"] for i in tag_list}
+
+
+def get_tags_for_rds_arn(session, rds_arn):
+    """Returns a dictionary of existing tags.
+    Args:
+        rds_arn (str): an RDS resource ARN.
+    Returns:
+        dict: A dictionary where tag names are keys and tag values are values.
+    """
+    return make_tag_dict(
+        session.list_tags_for_resource(ResourceName=rds_arn)["TagList"]
+    )
+
+
+def destroy_database(session, db_id, db_arn=None):
+    """Destroy the RDS db instance.
+    Args:
+        session (:class:`boto.rds2.layer1.RDSConnection`): The RDS api
+            connection where the database is located.
+        db_id (str): The RDS database instance identifier to destroy.
+    """
+    if db_arn is None:
+        description = get_database_description(session, db_id)
+        db_arn = description["DBInstanceArn"]
+
+    tags = get_tags_for_rds_arn(session, db_arn)
+
+    if tags.get("dbsnap-verify", "false") != "true":
```

Comment:
This is a good idea.

This is one of the instances where I'd go for a global constant, TAG_NAME and TAG_VALUE for "dbsnap-verify" and "true" (or w/e) -- this code is critical so I think it's best to fool-proff it against future edits.